Assets/_BForBoss/_Weapons/Scripts/Weapons/CrosshairBehaviour.cs

```diff
@@ -7,23 +8,74 @@ namespace Perigon.Weapons
     public class CrosshairBehaviour : MonoBehaviour
     {
         [SerializeField] private Sprite _defaultCrosshair = null;
-        private Image _crosshair = null;
+        [Resolve][SerializeField] private Image _crosshair = null;
 
-        private void Awake()
-        {
-            _crosshair = GetComponentInChildren<Image>();
-            SetDefaultCrosshair();
-        }
+        [Title("HitMarkers")]
+        [SerializeField] private float _hitMarkerStayOnScreenTime = 0.2f;
+        [Resolve][SerializeField] private Image _hitMarker = null;
 
+        private float _elapsedHitMarkerTime = 0;
+        private float _elapsedKillMarkerTime = 0;
+            
         public void SetDefaultCrosshair()
         {
             _crosshair.sprite = _defaultCrosshair;
         }
 
-        [Button]
         public void SetCrosshairImage(Sprite image)
         {
             _crosshair.sprite = image;
         }
+
+        public void ActivateHitMarker(bool isDead)
+        {
+            if (isDead)
+            {
+                _elapsedKillMarkerTime = _hitMarkerStayOnScreenTime;
+            }
+            
+            _elapsedHitMarkerTime = _hitMarkerStayOnScreenTime;
+        }
+
+        private void Update()
+        {
+            _elapsedHitMarkerTime -= Time.deltaTime;
+            _elapsedKillMarkerTime -= Time.deltaTime;
+            if (_elapsedKillMarkerTime > 0)
+            {
+                _hitMarker.color = Color.red;
+            } 
+            else if (_elapsedHitMarkerTime > 0)
+            {
+                _hitMarker.color = Color.white;
```

Comment:
Do you want to make these colors a serialized field variable, so we can change it from the inspector later down the line if need be

Reply:
It might not even be color in the future, these are almost placeholder. I could extract these out to be read-only variables?

Reply:
If its a placeholder, then I leave it up to you! Yeah maybe a Read-Only would be better?